### PR TITLE
Clock

### PR DESCRIPTION
**Pull request by alexander-ber**
Hi, Ameed.

Please check the HW.
I don't quite understand for what is the second bonus?

Thank's.

### CODE REVIEW

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 1]
```diff
@@ -34,4 +34,79 @@
...
+        this.milliseconds = (byte) milliseconds;
+    }
+
+	public void addHours(int hours) {
+		if(hours > 0 && hours < 1000000000 ) {
+			this.hours = (byte) ((this.hours + hours) % 24);
+		}
+	}
+	
+	public void addMinutes(int minutes) {
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java:64]: bug found: clock = 16:49:05.065 --> adding 30 minutes: 16:19:05.065 (should've been 17:19:05.065)

File: src/ameedf/jb8223/assignments/hw02/Clock.java [hunk 2]
```diff
@@ -34,4 +34,79 @@
...
+			this.milliseconds = (short) ((this.milliseconds + milliseconds) % 1000);
+		}
+	}
+	
+	public void print(boolean is12) {
+		String postfix = "";
+		int hoursToShow = hours;
+		if(is12) {
+			if(hours<=12) {
+				postfix = "PM";
```
Comment (ameedf) [inline src/ameedf/jb8223/assignments/hw02/Clock.java]: https://en.wikipedia.org/wiki/12-hour_clock

**Review by ameedf — COMMENTED** [review]
A good job
you have some bugs (gave you only an example) but a very clean code
Bonus 2: hint - use only one member variable

File: src/ameedf/jb8223/assignments/hw02/Clock.java (hunk 1 above)
Comment (alexander-ber) [reply src/ameedf/jb8223/assignments/hw02/Clock.java:64]: Yes, sorry, fixed.

Don't commit to production after midnight.. ))

File: src/ameedf/jb8223/assignments/hw02/Clock.java (hunk 2 above)
Comment (alexander-ber) [reply src/ameedf/jb8223/assignments/hw02/Clock.java]: My mistake.. fixed

I was sure that this works as I think and not how it really is.

**Comment by alexander-ber** [comment]
regarding bonus2 - thanks, I got it )) 
When I have time - I will send it.
Please check the bonus1.

Thanks

File: src/ameedf/jb8223/assignments/hw02/Clock.java (hunk 1 above)
Comment (ameedf) [reply src/ameedf/jb8223/assignments/hw02/Clock.java:64]: :)

**Comment by alexander-ber** [comment]
Hi Ameed,

Add bonus2 file.

**Pull request closed without merging**